### PR TITLE
Add disconnecting the client with arp when refreshing client list

### PR DESCRIPTION
**Pull request by blogcin**
nodogsplash does not automatically check client disconnects.

### CODE REVIEW

File: src/firewall.c [hunk 1]
```diff
@@ -165,7 +179,18 @@ fw_refresh_client_list(void)
 			now = time(NULL);
 			last_updated = cp1->counters.last_updated;
 			added_time = cp1->added_time;
-			if (last_updated +  (config->checkinterval * config->clienttimeout) <= now) {
+
+			if (is_arp_valid() && !arp_get(ip)) {
```
Comment (mwarning) [inline src/firewall.c]: arp_get() allocates memory that must be freed by the caller. This is a memory leak. :P

**Comment by mwarning** [comment]
@blogcin: this might slow down NDS a lot. For every entry, the whole arp table is iterated. /proc/arp is opened twice. Well, that can be improved.

The general idea looks fine.
@lynxis any thoughts on this one?

**Comment by lynxis** [comment]
/proc/net/arp get's garbage collected when reaching certain thresholds (lowest is 128).
So I would not like to rely on this to decide if this node got disconnected or not.

**Comment by lynxis** [comment]
The comment of the function is **wrong**. If we would still ping the client, we can try to look into the arp cache.
If we want to ping the client, we should ping the client using ARP rather than icmp.

**Comment by blogcin** [comment]
Thanks.

**Pull request closed without merging**